common/thanos/templates/ruler/ruler.yaml

```diff
@@ -0,0 +1,58 @@
+{{- $root := . }}
+{{- if and .Values.ruler.enabled (eq .Values.deployWholeThanos false) }}
+{{- range $name := coalesce .Values.names .Values.global.targets (list .Values.name) }}
+apiVersion: monitoring.coreos.com/v1
+kind: ThanosRuler
+metadata:
+  name: {{ include "thanos.name" (list $name $root) }}
+spec:
+  image: {{ include "thanosimage" $root }}
+  paused: {{ $.Values.ruler.paused }}
+  logLevel: {{ default "info" $.Values.ruler.logLevel }}
+{{- if $.Values.ruler.retention }}
+  retention: {{ $.Values.ruler.retention | quote }}
+{{- end }}
+{{- if $.Values.rbac.create }}
+  serviceAccountName: {{ include "serviceAccount.name" (list $name $root) }}
```

Comment:
Is this ruler specific? It's mapped in the overall values but we haven't used it before.

Reply:
Ok I've seen the label added to the ServiceAccount. Still would move this to the `ruler` section as it is very specific judging its label.